fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java

```diff
@@ -1076,36 +1113,53 @@ private void clearBackendTransactions(TransactionState transactionState) {
     }
 
     public void removeExpiredTxns(long currentMillis) {
+        List<Long> expiredTxnIds = Lists.newArrayList();
+        // delete expired txns
         writeLock();
         try {
-            while (!finalStatusTransactionStateDeque.isEmpty()) {
-                TransactionState transactionState = finalStatusTransactionStateDeque.getFirst();
-                if (transactionState.isExpired(currentMillis)) {
-                    finalStatusTransactionStateDeque.pop();
-                    clearTransactionState(transactionState);
-                    editLog.logDeleteTransactionState(transactionState);
-                    LOG.info("transaction [" + transactionState.getTransactionId() + "] is expired, remove it from transaction manager");
-                } else {
-                    break;
-                }
+            unprotectedRemoveExpiredTxns(currentMillis, expiredTxnIds, finalStatusTransactionStateDequeShort);
```

Comment:
I suggest that we should do some experiments to make sure log BatchRemoveTransactionsOperation performance when there are many txns expired as the same time? I doubt that too many txns expired would cause bad performance when write log for bdbje?

Reply:
We tested but writing a value of size 50MB to bdbje cost about 500ms.
And the number of txn here should be much smaller than it.
So I think we don't need to wrong about it.